File: app/src/main/java/com/rodolfonavalon/canadatransit/model/database/Operator.java
package com.rodolfonavalon.canadatransit.model.database;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import org.joda.time.DateTime;

import java.util.List;

import lombok.Getter;

public class Operator {

    @SerializedName("onestop_id") // TODO: primary key
    @Getter private String onestopId;

    @SerializedName("name")
    @Getter private String name;
    @SerializedName("website")
    @Getter private String website;
    @SerializedName("country")
    @Getter private String country;
    @SerializedName("state")
    @Getter private String state;
    @SerializedName("timezone")
    @Getter private String timezone;
    @SerializedName("created_at")
    @Getter private DateTime createdAt;
    @SerializedName("updated_at")
    @Getter private DateTime updatedAt;

    @Nullable
    @SerializedName("metro")
    @Getter private String metro;
    @Nullable
    @SerializedName("short_name")
    @Getter private String shortName;

    @SerializedName("tags")
    @Getter private Tags tags;
    @SerializedName("represented_in_feed_onestop_ids")
    @Getter private List<String> representedInFeedOneStopIds;

    private static class Tags {

        @SerializedName("agency_id")
        @Getter private String agencyId;
        @SerializedName("agency_lang")
        @Getter private String agencyLang;
        @SerializedName("agency_phone")
        @Getter private String agencyPhone;
    }
}

File: app/src/main/kotlin/com/rodolfonavalon/canadatransit/model/database/Operator.kt
package com.rodolfonavalon.canadatransit.model.database

import com.google.gson.annotations.SerializedName

import java.util.*

class Operator(
        @SerializedName("onestop_id") val onestopId: String, // TODO: primary key
        @SerializedName("name") val name: String,
        @SerializedName("website") val website: String,
        @SerializedName("country") val country: String,
        @SerializedName("state") val state: String,
        @SerializedName("timezone") val timezone: String,
        @SerializedName("created_at") val createdAt: Date,
        @SerializedName("updated_at") val updatedAt: Date,
        @SerializedName("metro") val metro: String?,
        @SerializedName("short_name") val shortName: String?,
        @SerializedName("tags") val tags: Tags,
        @SerializedName("represented_in_feed_onestop_ids") val representedInFeedOneStopIds: List<String>
) {

    inner class Tags(
            @SerializedName("agency_id") val agencyId: String,
            @SerializedName("agency_lang") val agencyLang: String,
            @SerializedName("agency_phone") val agencyPhone: String
    )
}
